types/src/v2/error.rs

```diff
@@ -21,18 +21,50 @@ pub struct JsonRpcError<'a> {
 pub struct JsonRpcErrorAlloc {
 	/// JSON-RPC version.
 	pub jsonrpc: TwoPointZero,
-	/// Error object.
-	pub error: ErrorCode,
+	/// JSON-RPC error object.
+	pub error: JsonRpcErrorObjectAlloc,
 	/// Request ID.
 	pub id: Id,
 }
 
 impl fmt::Display for JsonRpcErrorAlloc {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		write!(f, "{:?}: {}: {:?}", self.jsonrpc, self.error, self.id)
+		write!(f, "{:?}: {:?}: {:?}", self.jsonrpc, self.error, self.id)
 	}
 }
 
+/// JSON-RPC error object.
+#[derive(Debug, PartialEq, Clone, Deserialize)]
+#[serde(deny_unknown_fields)]
```

Comment:
Do we really need this?

Reply:
The reason is the huge error type which requires ownership and we return it the user.

The actual body is read and dropped there maybe we could have like `to_owned` or something for  `JsonRpcErrorObject`, it's quite annoying of to have two different types but perhaps we could tackle that in another PR